src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -79,8 +93,15 @@ private static RecordValue RecordFromJsonObject(JsonElement element)
             {
                 var name = pair.Name;
                 var value = pair.Value;
+                FormulaType fieldType = null;
 
-                var paValue = FromJson(value);
+                // if TryGetFieldType fails, fieldType is set to Blank
+                if (!recordType?.TryGetFieldType(name, out fieldType) != true)
```

Comment:
I suppose this should be `if (recordType?.TryGetFieldType(name, out fieldType) != true)`?

Reply:
Interestingly, if we set fieldtype to null, it will parse everything to the end instead of return a JsonUntypedObject which however, fits our code better

Reply:
This is best effort here
If we can check the type, let's do it
If the property name is unknown, let's not do it and accept it as it comes